ch12.md

```diff
@@ -0,0 +1,256 @@
+// [Task('hail the monarchy'), Task('smash the patriarchy')]
+```
+
+Here we read a bunch of files and end up with a useless array of tasks. How might we fork each one of these? It would be most agreeable if we could switch the types around to have `Task Error [String]` instead of `[Task Error String]`. That way, we'd have one future value holding all the results, which is much more amendable to our async needs than several future values arriving at their leisure.
+
+Here's one last example of a sticky situation:
+
+```js
+// getAttribute :: String -> Node -> Maybe String
+// $ :: Selector -> IO Node
+
+// getControlNode :: IO (Maybe (IO Node))
+const getControlNode = compose(map(map($)), map(getAttribute('aria-controls')), $)
+```
+
+Look at those `IO`s longing to be together. It'd be just lovely to `join` them, let them dance cheek to cheek, but alas a `Maybe` stands between them like a chaperone at prom. Our best move here would be to shift their positions next to one another, that way each type can be together at last and our signature can be simplified to `IO (Maybe Node)`.
+
+## Type Feng Shui
+
+The *Traversable* interface consists of two glorious functions: `sequence` and `traverse`.
+
+Let's rearrange our types using `sequence`:
+
+```js
+sequence(Array, Maybe(['the facts'])) // [Maybe('the facts')]
+sequence(Task.of, Map({a: Task.of(1), b: Task.of(2)})) // Task(Map({a: 1, b: 2}))
+sequence(IO.of, Right(IO.of('buckle my shoe'))) // IO(Right('buckle my shoe'))
+sequence(Either.of, [Right('wing')]) // Right(['wing'])
+sequence(Task.of, Left('wing')) // Task(Left('wing'))
+```
+
+See what has happened here? Our nested type gets turned inside out like a pair of leather trousers on a humid summer night. The inner functor is shifted to the outside and vice versa. It should be known that `sequence` is bit particular about its arguments. It looks like this:
+
+```js
+// sequence :: (Traversable t, Applicative f) => (a -> f a) -> t (f a) -> f (t a)
+const sequence = (point, x) => x.sequence(point)
+```
+
+Let's start with the second argument. It must be a *Traversable* holding an *Applicative*, which sounds quite restrictive, but just so happens to be the case more often than not. It is the `t (f a)` which gets turned into a `f (t a)`. Isn't that expressive? It's clear as day the two types do-si-do around each other. That first argument there is merely a crutch and only necessary in an untyped language. It is a type constructor (our *point*) provided so that we can invert map-reluctant types like `Left` - more on that in a minute.
+
+Using `sequence`, we can shift types around with the precision of a sidewalk thimblerigger. But how does it work? Let's look at how a type, say `Either`, would implement it:
+
+```js
+Right.prototype.sequence = function(point) {
+  return this.__value.map(Right)
+}
+```
+
+Ah yes, if our `__value` is a functor (it must be an applicative, in fact), we can simply `map` our constructor to leap frog the type.
```

Comment:
What about renaming `point` to `pure` ?

Reply:
Ya know, I thought about that. Perhaps the best one would be `of` since we've already covered Pointed.

Reply:
` of` might be a better idea indeed to maintain consistency with previous chapters. The laws from this chapter also refer to `of`.

:+1: